### PR TITLE
Use new connection for each email

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

**Review by kmshin1397 — APPROVED** [review]

**Comment by kmshin1397** [comment]
waiting on CI to merge

**Review by dannygoldstein — APPROVED** [review]

**Comment by kmshin1397** [comment]
@acrellin We need to make sure to call `send_email` only when `len(recipients) > 0` I think? Not sure how the email clients would respond to empty recipients but I assume it would probably throw an error.

**Review by kmshin1397 — APPROVED** [review]

File: skyportal/tests/frontend/test_sources.py [hunk 1]
```diff
@@ -484,7 +474,7 @@ def test_source_notification(driver, user, public_group, public_source):
     header = driver.wait_for_xpath("//header")
     ActionChains(driver).move_to_element(header).click().perform()
     driver.click_xpath("//button[@data-testid='sendNotificationButton']")
-    driver.wait_for_xpath("//*[text()='Notification queued up sucessfully']")
+    driver.wait_for_xpath("//*[text()[contains(., 'Invalid email service')]]")
```
Comment (kmshin1397) [inline skyportal/tests/frontend/test_sources.py:477]:
```suggestion
    driver.wait_for_xpath("//*[text()='Notification queued up sucessfully']")
```

**Comment by acrellin** [comment]
@kmshin1397 looks like the email notification test is failing on both Travis and GA

**Review by kmshin1397 — CHANGES_REQUESTED** [review]

**Comment by kmshin1397** [comment]
@acrellin Yeah just realized that since we now catch the empty recipients the test should expect a success notification. The change I suggest above should fix it

**Review by kmshin1397 — APPROVED** [review]

**Comment by kmshin1397** [comment]
green check on GA yay

**Review by dannygoldstein — APPROVED** [review]

**Pull request merged**